picard/util/pipe.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2022 skelly37
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+import concurrent.futures
+import os
+from typing import Optional
+
+from picard.const.sys import (
+    IS_MACOS,
+    IS_WIN,
+)
+
+
+if IS_WIN:
+    import win32pipe  # type: ignore
+    import win32file  # type: ignore
+    from pywintypes import error as WinApiError  # type: ignore
+
+
+class PipeError(Exception):
+    pass
+
+
+class PipeErrorInvalidArgs(PipeError):
+    MESSAGE = "ERROR: Pipe() args argument has to be iterable"
+
+    def __init__(self, t):
+        raise PipeError(f"{self.MESSAGE}, {type(t)} provied.")
+
+
+class PipeErrorNotFound(PipeError):
+    MESSAGE = "ERROR: Pipe doesn't exist."
+
+    def __init__(self):
+        raise PipeError(self.MESSAGE)
```

Comment:
Instead of re-raising `PipeError `, you should just call parent `__init__()`:

```python
def __init__(self):
    super().__init__(self.MESSAGE)
```

That's the usual approach.